### ISSUE TITLE
calwebb_spec2 cube_build failing for a NIRSpec IFU exposure

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#5244]
_Issue [JP-1642](https://jira.stsci.edu/browse/JP-1642) was created on JIRA by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse):_

The cube_build step in calwebb_spec2 is crashing with the following traceback when processing a NIRSpec IFU exposure:
```java
2020-08-13 09:10:21,989 - stpipe.Spec2Pipeline.cube_build - INFO - Number of bands in cube: 1
2020-08-13 09:10:21,989 - stpipe.Spec2Pipeline.cube_build - DEBUG - Working on data from g395h, f290lp
2020-08-13 09:10:21,989 - stpipe.Spec2Pipeline.cube_build - DEBUG - number of files 1
2020-08-13 09:10:21,990 - stpipe.Spec2Pipeline.cube_build - DEBUG - Environmental "STRICT_VALIDATION" cannot be found. Using default value of "False".
2020-08-13 09:10:21,997 - stpipe.Spec2Pipeline.cube_build - INFO - Looping over slices to determine cube size .. this takes a while
2020-08-13 09:11:37,888 - stpipe.Spec2Pipeline.cube_build - INFO - Cube Geometry:
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - INFO - axis#  Naxis  CRPIX    CRVAL      CDELT(arcsec)  Min & Max (xi, eta arcsec)
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - INFO - Axis 1    49  25.00  98.81605353   0.10000000  -2.45000004   2.45000004
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - INFO - Axis 2    51  26.00 -66.82729024   0.10000000  -2.55000004   2.55000004
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - INFO - axis#  Naxis  CRPIX    CRVAL      CDELT(microns)  Min & Max (microns)
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - INFO - Axis 3  1868   1.00   2.85994413   0.00066500   2.85961163   4.10183163
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - INFO - Cube covers grating, filter: g395h, f290lp
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - INFO - Output Name: jw00214054001_04108_00001_nrs1_s3d.fits
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - DEBUG - Working on Band defined by: g395h f290lp
2020-08-13 09:11:37,889 - stpipe.Spec2Pipeline.cube_build - DEBUG - Environmental "STRICT_VALIDATION" cannot be found. Using default value of "False".
2020-08-13 09:11:37,897 - stpipe.Spec2Pipeline.cube_build - INFO - Mapping each NIRSpec slice to sky; this takes a while for NIRSpec data
2020-08-13 09:12:53,753 - stpipe.Spec2Pipeline.cube_build - INFO - Time to transform pixels to output frame = 75.9 s
2020-08-13 09:23:42,012 - stpipe.Spec2Pipeline.cube_build - INFO - Time to set initial dq values = 648.3 s
2020-08-13 09:29:55,938 - stpipe.Spec2Pipeline.cube_build - INFO - Time to match file to ifucube = 373.9 s
2020-08-13 09:29:56,151 - stpipe.Spec2Pipeline.cube_build - INFO - Average # of holes/wavelength plane is < 1
2020-08-13 09:29:56,151 - stpipe.Spec2Pipeline.cube_build - INFO - Total # of holes for IFU cube is : 0
2020-08-13 09:29:56,154 - stpipe.Spec2Pipeline.cube_build - INFO - Time to find Cube Flux = 0.2 s
2020-08-13 09:29:56,222 - stpipe.Spec2Pipeline.cube_build - INFO - Number of wavelength planes removed with no data: 3735
Traceback (most recent call last):
  File "/Users/bushouse/JWST/softw/jwst/jwst/pipeline/calwebb_spec2.py", line 106, in process
    result = self.process_exposure_product(
  File "/Users/bushouse/JWST/softw/jwst/jwst/pipeline/calwebb_spec2.py", line 261, in process_exposure_product
    result_extra = self.cube_build(result)
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/Users/bushouse/JWST/softw/jwst/jwst/cube_build/cube_build_step.py", line 343, in process
    result = thiscube.build_ifucube()
  File "/Users/bushouse/JWST/softw/jwst/jwst/cube_build/ifu_cube.py", line 783, in build_ifucube
    ifucube_model = self.setup_final_ifucube_model(0)
  File "/Users/bushouse/JWST/softw/jwst/jwst/cube_build/ifu_cube.py", line 2101, in setup_final_ifucube_model
    self.crval3 = self.zcoord[remove_start]
IndexError: index 1868 is out of bounds for axis 0 with size 1868
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec2Pipeline':
    One or more products failed to process. Failing calibration.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/bushouse/JWST/softw/jwst/scripts/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/step.py", line 156, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/cmdline.py", line 346, in step_from_cmdline
    step.run(*positional)
  File "/Users/bushouse/JWST/softw/jwst/jwst/stpipe/step.py", line 392, in run
    step_result = self.process(*args)
  File "/Users/bushouse/JWST/softw/jwst/jwst/pipeline/calwebb_spec2.py", line 124, in process
    raise RuntimeError(
RuntimeError: One or more products failed to process. Failing calibration.
```
This happened using code from the latest master branch.

The data directory listed above contains the all the necessary inputs for testing (asn file and input rate products).

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1642?focusedCommentId=480628#comment-480628):_

A bit more information on why this set of exposures did not fail in the past. In PR4974 ( 0.17.0 pipeline version) the zero wavelength planes in an IFU cube (in the initial or final planes of the cube) were removed. Before that an IFU cube could have every wavelength plane full of zero data.

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-1642?focusedCommentId=480743#comment-480743):_

[Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse)
Currently if IFU cube contains no data for any wavelength plane then we get the error reported in this ticket. Should I change this and report a log.error statement and then return to the main cube_build_step and set skip for cube_build so it exists nicely ? The extract_1d step would them probably error out.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1642?focusedCommentId=498727#comment-498727):_

Fixed by #5295
